File: Autos-stock-manager-services/src/main/java/org/autostock/dtos/VoitureCreateDto.java
package org.autostock.dtos;

import lombok.Getter;
import lombok.Setter;
import org.autostock.enums.CategorieVoiture;
import org.autostock.enums.StatutVoiture;

import java.math.BigDecimal;
import java.time.LocalDate;

@Getter @Setter
public class VoitureCreateDto {
    private Long idModele;
    private Long idFournisseur; // optionnel
    private Integer annee;
    private String couleur;
    private String vin;
    private BigDecimal prixAchat;
    /** Prix affiche sur la vitrine. Le prix de vente reel vient de la vente. */
    private BigDecimal prixDemande;
    private Long kilometrage;
    StatutVoiture statut;
    CategorieVoiture categorie;
    LocalDate dateEntreeStock;
    Boolean creerMouvementEntree;
    // getters / setters
}

File: autos-stock-client/src/app/models/VoitureCreateDto.ts
import {StatutVoiture} from "./enums/StatutVoiture";
import {CategorieVoiture} from "./enums/CategorieVoiture";

export interface VoitureCreateDto {
  marque: string;
  modele: string;
  annee: number;
  vin?: string;
  couleur?: string;
  kilometrage?: number;
  prixAchat?: number;
  /** Prix affiche sur la vitrine. Le prix de vente reel vient de la vente. */
  prixDemande?: number;
  statut?: StatutVoiture;
  categorie?: CategorieVoiture;
  idFournisseur?: number;
  dateEntreeStock?: string;   // 'YYYY-MM-DD'
  creerMouvementEntree?: boolean; // true => crée un mouvement "ENTREE_STOCK"
}
